### ISSUE TITLE
Forecast of n timestamps

### ISSUE THREAD

**Issue by pedrozamboni** [eduardoAcunaEspinoza/Hy2DL#45]
Hi,
Can I use the  lstmmdn model to predict a n number of streamflow values? 
Or, should I use another model? 
Best

**Comment by eduardoAcunaEspinoza** [comment]
Hi Pedro, 

Yes you can do that. MDN is based on cudalstm which supports predicting multiple values in hindcast and in forecast 

In your case, would you like to predict n values for hindcast or forecast? I could upload an example of how to do that.

**Comment by pedrozamboni** [comment]
Hi Eduardo,

I would like to use predict n values on the forecast. I`m current working on adding a new dataset for Brazil (https://zenodo.org/records/4070147) and would like to train a model for short term forecast (maybe 7 to 10 days) and a second for seasonal forecasting (perhaps aggregating daily to mean flow data and predicting the next 4 to 6 months)

**Comment by eduardoAcunaEspinoza** [comment]
Ok understood. Yes, as I mentioned before, MDN is based on cudalstm, so you can use the MDN network in a forecast model (https://hy2dl.readthedocs.io/en/latest/usage/modelzoo/cudalstm.html#forecast-lstm). I will upload an example of that in the next few days.

It would be great if you add the support class for CAMELS Brazil.

**Comment by pedrozamboni** [comment]
HI Eduardo, 
Thank yoou. Yeap, I also working on adding a CAMELS BR class on the datazoo.
I can share with you when itts done.
Best

**Comment by eduardoAcunaEspinoza** [comment]
Hi Pedro,

I will try to upload the example next week, because right now the cluster where we try things is down. However, here is an example of what the config would look like so you can start trying it out. 

In this example we train and test with "pseudoforecast"  meaning we used observed data as a proxy of numerical weather forecast. The library also supports using real forecasts, but to include an example of that I need more time. 

Hope this helps. 

```yaml
experiment_name: ForecastMDN_CAMELS_DE
path_data: ../data/CAMELS_DE
path_entities: ../data/basin_id/basins_camels_de_1582.txt
dynamic_input:
- precipitation_mean
- precipitation_stdev
- radiation_global_mean
- temperature_min
- temperature_max
pseudo_forecast_input:
- precipitation_mean
- precipitation_stdev
- radiation_global_mean
- temperature_min
- temperature_max
target:
- discharge_spec
static_input:
- area
- elev_mean
- clay_0_30cm_mean
- sand_0_30cm_mean
- silt_0_30cm_mean
- artificial_surfaces_perc
- agricultural_areas_perc
- forests_and_seminatural_areas_perc
- wetlands_perc
- water_bodies_perc
- p_mean
- p_seasonality
- frac_snow
- high_prec_freq
- low_prec_freq
- high_prec_dur
- low_prec_dur
training_period:
- '1970-10-01'
- '1999-12-31'
validation_period:
- '1965-10-01'
- '1970-09-30'
testing_period:
- '2000-01-01'
- '2020-12-31'
seq_length_hindcast: 360
seq_length_forecast: 5
predict_last_n: 5
dynamic_embedding:
  hiddens:
  - 5
hidden_size: 128
batch_size_training: 256
batch_size_evaluation: 1024
epochs: 20
dropout_rate: 0.4
learning_rate: 0.0005
validate_every: 1
device: gpu
num_workers: 4
dataset: camels_de
model: lstmmdn
num_mixture_components: 3
distribution: laplacian
loss : nll
validation_metric:
- nse
- nll
initial_forget_bias: 3.0
random_seed: 110

**Comment by pedrozamboni** [comment]
Hi Eduardo, 

Thank you. Hope I can start some try outs next week. 
Hope to see an example of real forecast in the near future. 

Best
